ddp-workspace/projects/ddp-dsm-ui/src/app/guards/checkAuth.guard.ts

```diff
@@ -15,16 +15,16 @@ export class CheckAuthGuard implements CanLoad {
   }
 
   canLoad(route: Route, segments: UrlSegment[]): boolean | UrlTree {
-    const selectedRealm = localStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+    const selectedRealm = sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM);
 
     if (!this.auth.authenticated() && !selectedRealm) {
       return true;
     }
     else if (this.auth.authenticated() && !selectedRealm) {
-      this.auth.logout();
+      this.auth.sessionLogout();
```

Comment:
This path leaves the login token because removing it will log other users out.  What we need here is to get the user back to the study pickList.